core-web/libs/template-builder/src/lib/directives/container-options.directive.ts

```diff
@@ -0,0 +1,85 @@
+import { Observable, Subscription } from 'rxjs';
+
+import { Directive, OnDestroy, OnInit, Optional, Self } from '@angular/core';
+
+import { Dropdown } from 'primeng/dropdown';
+
+import { debounceTime, map, switchMap, tap } from 'rxjs/operators';
+
+import { DotContainersService } from '@dotcms/data-access';
+import { DotContainerEntity, DotDropdownSelectOption } from '@dotcms/dotcms-models';
+
+const DEFAULT_LABEL_NAME_INDEX = 'label';
+const DEFAULT_VALUE_NAME_INDEX = 'value';
+
+/**
+ * Directive to set a default configuration of Dropdown or MultiSelect (PrimeNG) and translate the label of the options
+ *
+ * @export
+ * @class DotDropdownDirective
+ */
+@Directive({
+    selector: 'p-dropdown[dotcmsContainerOptions]',
+    standalone: true
+})
+export class ContainerOptionsDirective implements OnInit, OnDestroy {
+    private readonly control: Dropdown;
+    private readonly maxOptions = 10;
+    private defaultOptions: Array<DotDropdownSelectOption<DotContainerEntity>>;
+    private onFilterSubscription: Subscription;
+
+    constructor(
+        @Optional() @Self() private readonly primeDropdown: Dropdown,
+        private readonly dotContainersService: DotContainersService
+    ) {
+        this.control = this.primeDropdown;
+
+        if (this.control) {
+            this.control.optionLabel = DEFAULT_LABEL_NAME_INDEX;
+            this.control.optionValue = DEFAULT_VALUE_NAME_INDEX;
+            this.control.showClear = this.control instanceof Dropdown ? true : false;
+        } else {
+            console.warn(
+                'ContainerOptionsDirective is for use with PrimeNg Dropdown or MultiSelect'
+            );
+        }
+    }
+
+    ngOnInit() {
+        this.fetchContainerOptions('').subscribe((options) => {
+            this.control.options = this.control.options || options; // avoid overwriting if they were already set
+            this.defaultOptions = options;
+        });
+        this.onFilterSubscription = this.control.onFilter
+            .pipe(
+                debounceTime(500),
+                switchMap((event: { filter: string }) => {
+                    return this.fetchContainerOptions(event.filter);
+                }),
+                tap((options) => this.setOptions(options))
+            )
+            .subscribe();
```

Comment:
Wanted to confirm with the team if using a subscription is a good idea. I know we usually try to push for async pipe, but think that's not a possibility from a directive.

Reply:
When we cannot make use of `Async` pipe, we prefer to do the `takeUntil(destroy$)` approach, that way if you have a  more observers it destroy everyone of them here is an example:

```javascript
private destroy$: Subject<boolean> = new Subject<boolean>();

 ngOnInit() {
        observableFromEvent(this.searchInput.nativeElement, 'keyup')
            .pipe(debounceTime(500), takeUntil(this.destroy$))
            .subscribe((keyboardEvent: Event) => {
                this.filterConfigurations(keyboardEvent.target['value']);
            });
    }
    
ngOnDestroy(): void {
        this.destroy$.next(true);
        this.destroy$.complete();
    }
```

That way when `this.destroy$.complete()` is called, all the observers with `takeUntil(destroy$)` will be destroyed